### PR TITLE
Define W3DModelDraw classes.

### PR DESCRIPTION
**Pull request by jonwil**
Add some things required to implement them.

### CODE REVIEW

File: src/game/client/draw/w3dmodeldraw.h [hunk 1]
```diff
@@ -0,0 +1,390 @@
...
+    int m_mode;
+    std::vector<ParticleSysBoneInfo> m_ParticleSysBones;
+    uint64_t m_transition;
+    float m_animationSpeedFactorMin;
+    float m_animationSpeedFactorMax;
+    std::map<NameKeyType, PristineBoneInfo> m_boneMap;
+    TurretInfo m_turretInfo[2];
+    std::vector<WeaponBarrelInfo> m_weaponBarrelInfoVec[3];
+    bool m_hasWeaponBone[3];
+    char m_validateFlags;
```
Comment (tomsons26) [inline src/game/client/draw/w3dmodeldraw.h]: seems actually called m_validStuff... & 1 is PRISTINE_BONES_VALID

File: src/game/client/draw/w3dmodeldraw.h [hunk 2]
```diff
@@ -0,0 +1,390 @@
...
+    Utf8String m_weaponLaunchBoneName[3];
+    Utf8String m_weaponHideShowBoneName[3];
+    std::vector<W3DAnimationInfo> m_animations;
+    NameKeyType m_transitionKey;
+    NameKeyType m_allowToFinishKey;
+    int m_flags;
+    int m_iniReadFlags;
+    int m_mode;
+    std::vector<ParticleSysBoneInfo> m_ParticleSysBones;
+    uint64_t m_transition;
```
Comment (tomsons26) [inline src/game/client/draw/w3dmodeldraw.h:183]: don't see this anywhere....

File: src/game/client/draw/w3dmodeldraw.h (hunk 1 above)
Comment (jonwil) [reply src/game/client/draw/w3dmodeldraw.h]: fixed.

File: src/game/client/draw/w3dmodeldraw.h (hunk 2 above)
Comment (jonwil) [reply src/game/client/draw/w3dmodeldraw.h:183]: Look at 00620499 in wb, the function 00633260 returns a uint64_t that is then put into m_transition

**Review by tomsons26 — APPROVED** [review]

**Pull request merged**